### PR TITLE
Fearure/bngp 1099

### PR DESCRIPTION
**Pull request by reneli5**

### CODE REVIEW

File: packages/webapp/src/utils/constants.js [hunk 1]
```diff
@@ -118,6 +118,9 @@ const NEED_MANAGEMENT_PLAN = 'land/need-management-plan'
 const NEED_LEGAL_AGREEMENT = 'land/need-legal-agreement'
 const REFERER = 'referer'
 const YES = 'yes'
+const LAND_OWNER_EMAIL = 'land/email'
```
Comment (teddmason) [inline packages/webapp/src/utils/constants.js:121]: Can var name match the route? Makes it easier for writing tests etc

File: packages/webapp/src/utils/constants.js [hunk 2]
```diff
@@ -118,6 +118,9 @@ const NEED_MANAGEMENT_PLAN = 'land/need-management-plan'
...
+const CORRECT_OWNER_EMAIL = 'land/correct-email'
```
Comment (teddmason) [inline packages/webapp/src/utils/constants.js:122]: again match var name to route

File: packages/webapp/src/utils/register-task-list.js [hunk 3]
```diff
@@ -9,7 +9,8 @@ const registerTaskList = {
           title: 'Add your details',
           status: constants.DEFAULT_REGISTRATION_TASK_STATUS,
           completedTaskUrl: constants.routes.CHECK_YOUR_DETAILS,
-          startTaskUrl: constants.routes.NAME
+          startTaskUrl: constants.routes.NAME,
+          id: 'add-your-details'
```
Comment (teddmason) [inline packages/webapp/src/utils/register-task-list.js:13]: What are these ID additions on this script? I don't think they're related to BNGP-1099

File: packages/webapp/src/routes/__tests__/land/register-land-task-list.spec.js [hunk 4]
```diff
@@ -2,7 +2,8 @@ import { submitGetRequest } from '../helpers/server.js'
 import constants from '../../../utils/constants'
 import { getRegistrationTasks } from '../../../utils/helpers'
 
-const url = constants.routes.REGISTER_LAND_TASK_LIST
+// const url = constants.routes.REGISTER_LAND_TASK_LIST
```
Comment (teddmason) [inline packages/webapp/src/routes/__tests__/land/register-land-task-list.spec.js:5]: commented code

File: packages/webapp/src/routes/__tests__/land/register-land-task-list.spec.js [hunk 5]
```diff
@@ -2,7 +2,8 @@ import { submitGetRequest } from '../helpers/server.js'
...
+const url = '/land/register-land-task-list'
```
Comment (teddmason) [inline packages/webapp/src/routes/__tests__/land/register-land-task-list.spec.js:6]: This string needs switching to the constants.routes (which was commented out)

File: packages/webapp/src/routes/__tests__/land/role.spec.js [hunk 6]
```diff
@@ -19,7 +19,7 @@ describe(url, () => {
     it('Should continue journey if role selected', async () => {
       postOptions.payload.role = 'Ecologist'
       const res = await submitPostRequest(postOptions)
-      expect(res.headers.location).toEqual('/land/check-your-details')
+      expect(res.headers.location).toEqual('/land/email')
```
Comment (teddmason) [inline packages/webapp/src/routes/__tests__/land/role.spec.js:22]: better to use the constants.routes.EMAIL

File: packages/webapp/src/routes/land/correct-email.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+const handlers = {
+  get: async (_request, h) => {
+    const emailAddress = _request.yar.get(constants.redisKeys.LAND_OWNER_EMAIL)
+    return h.view(constants.views.CORRECT_OWNER_EMAIL, {
+      emailAddress
+    })
+  },
+  post: async (request, h) => {
+    let vewPage
```
Comment (teddmason) [inline packages/webapp/src/routes/land/correct-email.js:12]: spelling of vewPage

File: packages/webapp/src/routes/land/correct-email.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        vewPage = h.redirect(constants.routes.CHECK_YOUR_DETAILS)
+      } else {
+        vewPage = h.view(constants.views.CORRECT_OWNER_EMAIL, { errorMessage: emailStatus.err[0].text, selected: true })
+      }
+    }
+    return vewPage
+  }
+}
+
+const setEmailSetails = request => {
```
Comment (teddmason) [inline packages/webapp/src/routes/land/correct-email.js:29]: method spelling?

File: packages/webapp/src/routes/land/correct-email.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+  post: async (request, h) => {
+    let vewPage
+    if (request.payload.correctEmail === 'yes') {
+      setEmailSetails(request)
+      vewPage = h.redirect(constants.routes.CHECK_YOUR_DETAILS)
+    } else {
+      const emailStatus = validateEmail(request.payload.emailAddress)
+      if (!emailStatus) {
+        setEmailSetails(request)
+        vewPage = h.redirect(constants.routes.CHECK_YOUR_DETAILS)
```
Comment (teddmason) [inline packages/webapp/src/routes/land/correct-email.js:20]: I think on `/correct-email` page you need to check for referer and redirect if set.

File: packages/webapp/src/views/land/check-and-submit.html [hunk 10]
```diff
@@ -50,6 +50,23 @@ <h2 class="govuk-heading-l">Your details</h2>
...
+              key: {
+                text: "Email address"
+              },
+              value: {
+                text: application.applicant.email
+              },
+              actions: {
+                items: [
+                  {
+                    href: routes.CORRECT_OWNER_EMAIL,
```
Comment (teddmason) [inline packages/webapp/src/views/land/check-and-submit.html:64]: According to Dave Hunter's comment when editing the email address it should take you to the /email route not the /correct-email route https://eaflood.atlassian.net/browse/BNGP-1099?focusedCommentId=453117

File: packages/webapp/src/views/land/check-land-boundary-details.html [hunk 11]
```diff
@@ -23,48 +23,63 @@
...
+                                    text: landBoundaryFileName
+                                },
                                 actions: {
                                 items: [
                                     {
                                         href: '/land/check-land-boundary-file',
-                                        text: 'Change'
+                                        text: 'Change',
+                                        attributes: {
+                                            id: "check-land-boundary-file"
```
Comment (teddmason) [inline packages/webapp/src/views/land/check-land-boundary-details.html:37]: If these IDs are meant to be included in this pull request please add the ticket ID to the PR description and title. As they're not on BNGP-1099

**Review by teddmason — CHANGES_REQUESTED** [review]
Few things need sorting out, including description and title of the PR to include the ID ticket.

File: packages/webapp/src/utils/constants.js [hunk 12]
```diff
@@ -234,7 +237,9 @@ const redisKeys = {
   REGISTRATION_TASK_DETAILS,
   LAND_OWNERSHIP_KEY,
   METRIC_UPLOADED_ANSWER,
-  REFERER
+  REFERER,
+  LAND_OWNER_EMAIL,
```
Comment (teddmason) [inline packages/webapp/src/utils/constants.js:241]: Rediskeys normally have a distinct key rather than using a route. for LAND_OWNER_EMAIL and CONFIRM_OWNER_EMAIL

**Review by pwadmore-ea — CHANGES_REQUESTED** [review]

File: packages/webapp/src/utils/register-task-list.js (hunk 3 above)
Comment (reneli5) [reply packages/webapp/src/utils/register-task-list.js:13]: Yes, they are not related to this ticket. Shaun wanted the ids for automated testing. I included them in this ticker.

File: packages/webapp/src/views/land/check-and-submit.html (hunk 10 above)
Comment (reneli5) [reply packages/webapp/src/views/land/check-and-submit.html:64]: The implementation is in alignment with Dave's comment

File: packages/webapp/src/routes/land/correct-email.js (hunk 7 above)
Comment (teddmason) [reply packages/webapp/src/routes/land/correct-email.js]: miss-spelling still remains

File: packages/webapp/src/routes/__tests__/land/role.spec.js [hunk 13]
```diff
@@ -1,6 +1,7 @@
 import { submitGetRequest, submitPostRequest } from '../helpers/server.js'
 import constants from '../../../utils/constants'
-const url = constants.routes.ROLE
+// const url = constants.routes.ROLE
```
Comment (teddmason) [inline packages/webapp/src/routes/__tests__/land/role.spec.js:3]: commented code needs reinstating with constant var, and string removing.

**Review by pwadmore-ea — APPROVED** [review]

**Review by teddmason — APPROVED** [review]
